pymatgen/util/coord_cython.pyx

```diff
@@ -178,6 +179,8 @@ def pbc_shortest_vectors(lattice, fcoords1, fcoords2, mask=None, return_d2=False
                 for l in range(3):
                     vs[i, j, l] = 1e20
 
+    if not (n_pbc == 3):
+        free(&frac_im[0,0])
```

Comment:
These changes don't look like they should affect performance. Not sure how fast the `free` operation is but considering we're currently leaking memory, _some_ slowdown is definitely preferable.

Reply:
Just a minor pt - find_pbc_vector is very very performance critical (this is why it is in cython). I am willing to lose < 10% performance for mem, but if it is a 10x diff (this is not difficult to imagine), we need to consider the tradeoff. Basically all operations involving bond distances will slow down. Since the performance loss seems minimal, this is ok.